Current master breaks ``make clean``

### Describe the bug

We can not execute ``make clean`` anymore on current Sphinx HEAD

```
$ make html && make clean
Sphinx v8.2.0+/de24fd0b7 en cours d'exécution
chargement des traductions [en]... fait
chargement de l'environnement pickled... fait
construction en cours [mo] : cibles périmées pour les fichiers po 0
écriture... 
construction [html] : cibles périmées pour les fichiers sources 0
mise à jour de l'environnement : 0 ajouté(s), 0 modifié(s), 0 supprimé(s)
lecture des sources... 
recherche des fichiers périmés... aucun résultat trouvé
aucune cible n'est périmée.
La compilation a réussi.

Les pages HTML sont dans _build/html.
Removing everything under '_build'...
Traceback (most recent call last):
  File "/path/to/work/.venv312/bin/sphinx-build", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/path/to/sphinx/sphinx/cmd/build.py", line 561, in main
    return make_mode.run_make_mode(argv[1:])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/path/to/sphinx/sphinx/cmd/make_mode.py", line 222, in run_make_mode
    return getattr(make, run_method)()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/path/to/sphinx/sphinx/cmd/make_mode.py", line 90, in build_clean
    rmtree(self.build_dir_join(item))
  File "/path/to/sphinx/sphinx/util/osutil.py", line 243, in rmtree
    os.remove(path)
FileNotFoundError: [Errno 2] No such file or directory: '_build/_build/html'
make: *** [Makefile:20: clean] Error 1
```

### How to Reproduce

sphinx-quickstart then ``make html && make clean`` .  Install first Sphinx master in a venv.

### Environment Information

```text
Platform:              darwin;
Python version:        3.12.3 (v3.12.3:f6650f9ad7, Apr  9 2024, 08:18:48) [Clang 13.0.0 (clang-1300.0.29.30)])
Python implementation: CPython
Sphinx version:        8.2.0+/de24fd0b7
Docutils version:      0.21.2
Jinja2 version:        3.1.4
Pygments version:      2.18.0
```


### Sphinx extensions

```python
none
```


### Additional context

First bad commit is reported by `git bisect` to be de24fd0b76f6d86672a721f8c130642b0be4bcf0

I am on macOS, too old to reveal which one here publicly (I don't want to be attacked by hackers).

--- a/sphinx/cmd/make_mode.py
+++ b/sphinx/cmd/make_mode.py
@@ -86,7 +86,7 @@ class Make:
             print("Error: '%s' directory contains source directory!" % self.build_dir)
             return 1
         print("Removing everything under '%s'..." % self.build_dir)
-        for item in build_dir.iterdir():
+        for item in self.build_dir.iterdir():
             rmtree(item)
         return 0
 

--- a/sphinx/cmd/make_mode.py
+++ b/sphinx/cmd/make_mode.py
@@ -86,8 +86,8 @@ class Make:
             print("Error: '%s' directory contains source directory!" % self.build_dir)
             return 1
         print("Removing everything under '%s'..." % self.build_dir)
-        for item in self.build_dir.iterdir():
-            rmtree(self.build_dir_join(item))
+        for item in build_dir.iterdir():
+            rmtree(item)
         return 0
 
     def build_help(self) -> None:

